From a644bc43ca53450d8ed9ea083b6bd091a000a7fd Mon Sep 17 00:00:00 2001
From: Angela Czubak <angela.czubak@cavium.com>
Date: Thu, 24 Jan 2019 16:16:19 -0800
Subject: [PATCH 0919/1051] soc: octeontx2: fix cgx_lmac_rx_tx_enable

Do not disable or enable the LMAC itself, only receiving or transmitting.

Change-Id: I7b874466d18f107de90b028ed2a68c9160a48938
Signed-off-by: Angela Czubak <angela.czubak@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/2707
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Chandrakala Chavva <Chandrakala.Chavva@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/cgx.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2/cgx.c b/drivers/soc/marvell/octeontx2/cgx.c
index f4c203308614..c3e89ce41769 100644
--- a/drivers/soc/marvell/octeontx2/cgx.c
+++ b/drivers/soc/marvell/octeontx2/cgx.c
@@ -390,9 +390,9 @@ int cgx_lmac_rx_tx_enable(void *cgxd, int lmac_id, bool enable)
 
 	cfg = cgx_read(cgx, lmac_id, CGXX_CMRX_CFG);
 	if (enable)
-		cfg |= CMR_EN | DATA_PKT_RX_EN | DATA_PKT_TX_EN;
+		cfg |= DATA_PKT_RX_EN | DATA_PKT_TX_EN;
 	else
-		cfg &= ~(CMR_EN | DATA_PKT_RX_EN | DATA_PKT_TX_EN);
+		cfg &= ~(DATA_PKT_RX_EN | DATA_PKT_TX_EN);
 	cgx_write(cgx, lmac_id, CGXX_CMRX_CFG, cfg);
 	return 0;
 }
-- 
2.17.1

